src/snowflake/connector/network.py

```diff
@@ -265,8 +266,9 @@ def get_connection(
                 )
             proxy_manager = self.proxy_manager_for(proxy)
 
-            # Add Host to proxy header SNOW-232777
-            proxy_manager.proxy_headers["Host"] = parsed_url.hostname
+            if not isinstance(proxy_manager, SOCKSProxyManager):
+                # Add Host to proxy header SNOW-232777
+                proxy_manager.proxy_headers["Host"] = parsed_url.hostname
```

Comment:
I may not have the best knowledge of proxy, but just to be defensive + general and get things work, I will probably go with:

```
try:
    proxy_manager.proxy_headers["Host"] = parsed_url.hostname
except AttributeError:
    # log that proxy_manager not having attribute proxy_headers
```

@sfc-gh-mkeller do you have any knowledge about this?

Reply:
@stan-janssen , just want to know how you think about my approach?

Reply:
Sure, that would work!